homeassistant/components/ryse/cover.py

```diff
@@ -0,0 +1,116 @@
+"""Support for RYSE Smart Shades via BLE."""
+
+import logging
+from typing import Any
+
+from ryseble.device import RyseBLEDevice
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up RYSE Smart Shade cover from a config entry."""
+    device = RyseBLEDevice(entry.unique_id)
+    async_add_entities([RyseCoverEntity(device, entry)])
+
+
+class RyseCoverEntity(CoverEntity):
+    """Representation of a RYSE Smart Shade BLE cover entity."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    def __init__(self, device: RyseBLEDevice, config_entry: ConfigEntry) -> None:
+        """Initialize the Smart Shade cover entity."""
+        self._device = device
+        self._config_entry = config_entry
+        self._attr_unique_id = f"{device.address}_cover"
+        self._current_position: int | None = None
+        self._attr_is_closed: bool | None = None
+        self._device.update_callback = self._update_position
```

Comment:
This should be set in `async_added_to_hass` and cleaned up properly via `async_will_remove_from_hass` or using `self.async_on_remove`